internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -426,6 +443,91 @@ func (r *ClusterDeploymentsReconciler) updateInstallConfigOverrides(ctx context.
 	return nil
 }
 
+func (r *ClusterDeploymentsReconciler) syncManifests(ctx context.Context, cluster *common.Cluster,
+	clusterDeployment *hivev1.ClusterDeployment, alreadyCreatedManifests models.ListManifests) error {
+
+	r.Log.Debugf("Going to sync list of given with already created manifests")
+
+	manifestsFromConfigMap, err := r.getClusterDeploymentManifest(ctx, clusterDeployment)
+	if err != nil {
+		return err
+	}
+
+	// delete all manifests that are not part of configmap
+	// skip errors
+	for _, manifest := range alreadyCreatedManifests {
+		if _, ok := manifestsFromConfigMap[manifest.FileName]; !ok {
+			r.Log.Infof("Deleting cluster deployment %s manifest %s", cluster.KubeKeyName, manifest.FileName)
+			_ = r.Manifests.DeleteClusterManifestInternal(ctx, operations.DeleteClusterManifestParams{
+				ClusterID: *cluster.ID,
+				FileName:  manifest.FileName,
+				Folder:    swag.String(models.ManifestFolderOpenshift),
+			})
+		}
+	}
+
+	// create/update all manifests provided by configmap data
+	for filename, manifest := range manifestsFromConfigMap {
+		r.Log.Infof("Creating cluster deployment %s manifest %s", cluster.KubeKeyName, filename)
+		_, err := r.Manifests.CreateClusterManifestInternal(ctx, operations.CreateClusterManifestParams{
+			ClusterID: *cluster.ID,
+			CreateManifestParams: &models.CreateManifestParams{
+				Content:  swag.String(base64.StdEncoding.EncodeToString([]byte(manifest))),
+				FileName: swag.String(filename),
+				Folder:   swag.String(models.ManifestFolderOpenshift),
+			}})
+		if err != nil {
+			r.Log.WithError(err).Errorf("Failed to create cluster deployment %s manifest %s", cluster.KubeKeyName, filename)
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *ClusterDeploymentsReconciler) getClusterDeploymentManifest(ctx context.Context, clusterDeployment *hivev1.ClusterDeployment) (map[string]string, error) {
+	configuredManifests := &corev1.ConfigMap{}
+	configuredManifests.Data = map[string]string{}
+	// get manifests from configmap if we have reference for it
+	if clusterDeployment.Spec.Provisioning.ManifestsConfigMapRef != nil {
+		err := r.Get(ctx, types.NamespacedName{Namespace: clusterDeployment.Namespace,
+			Name: clusterDeployment.Spec.Provisioning.ManifestsConfigMapRef.Name}, configuredManifests)
+		if err != nil {
+			r.Log.WithError(err).Errorf("Failed to get configmap %s in %s",
+				clusterDeployment.Spec.Provisioning.ManifestsConfigMapRef.Name, clusterDeployment.Namespace)
+			return nil, err
+		}
+	}
+	return configuredManifests.Data, nil
+}
+
+func (r *ClusterDeploymentsReconciler) addCustomManifests(ctx context.Context, clusterDeployment *hivev1.ClusterDeployment,
+	cluster *common.Cluster) error {
+
+	// added just in case we it will run after cluster will be in the middle of installation
+	if !common.IsInPreInstallationState(cluster) {
```

Comment:
This seems redundent, this method is called only when:
``` ready, err := r.isReadyForInstallation(ctx, clusterDeployment, cluster)```
returns ready=True

Reply:
yeah, sounds like